### PR TITLE
Refactor/ether swr to wagmi [PART 1]

### PR DESCRIPTION
**Pull request by Kenny-Gin1**
# Description

This is the first part of three to refactoring our hooks to use wagmi completely. I added some tests to it as well.

- Refactoring and moving all the hooks to `hooks/Guilds/ens`
- Adding tests for all the hooks (except `useENSAvatar`)

TODO: Add tests to `useENSAvatar` after further refactoring. 

Closes # https://github.com/DXgovernance/DAVI/issues/240

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Manually and unit tests.

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any UI changes have been tested and made responsive for mobile views

### CODE REVIEW

**Comment by Filipv95** [comment]
Hey @Kenny-Gin1, is there a way to test this part now?

**Comment by Kenny-Gin1** [comment]
> 

Hey! Yeah, this is another regression test in order. If nothing unexpectedly breaks - then it should be a go 👍

**Comment by Kenny-Gin1** [comment]
> Hey @Kenny-Gin1, is there a way to test this part now?

Hey! @Filipv95 For sure, this is another regression test in order. If nothing breaks unexpectedly then this should be fine!

File: src/hooks/Guilds/ens/useENS.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { isAddress } from '../../../../utils';
+import { isAddress } from '../../../utils';
```
Comment (dcrescimbeni) [inline src/hooks/Guilds/ens/useENS.ts:1]: Should we use an absolute path?

File: src/hooks/Guilds/ens/useENSAvatar.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
 import { useMemo } from 'react';
-import { resolveUri } from '../../../../utils/url';
+import { resolveUri } from '../../../utils/url';
```
Comment (dcrescimbeni) [inline src/hooks/Guilds/ens/useENSAvatar.ts:2]: Same as above

**Comment by dcrescimbeni** [comment]
Everything looks good! Left some minor comments about absolute paths.

**Comment by Filipv95** [comment]
Hey @Kenny-Gin1 tnx, I've finished regression testing on this branch, every thing seems good.

Noticed one thing tho - sometimes I am unable to vote on a Proposal from second account. I'm not able to reproduce it every time, perhaps we should check this. We have same behavior on `development` branch so It's not problem with this branch.

**Comment by Kenny-Gin1** [comment]
> Hey @Kenny-Gin1 tnx, I've finished regression testing on this branch, every thing seems good.
> 
> Noticed one thing tho - sometimes I am unable to vote on a Proposal from second account. I'm not able to reproduce it every time, perhaps we should check this. We have same behavior on `development` branch so It's not problem with this branch.

Good to know! I think make an issue of that and let's discuss it on Monday's sprint?

**Review by dcrescimbeni — DISMISSED** [review]

**Comment by Filipv95** [comment]
Hey @Kenny-Gin1 sry, I'll not be able to attend. But I will log the issue of course!

**Review by Filipv95 — APPROVED** [review]

File: src/old-components/Guilds/AddressButton/index.tsx [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { isDesktop } from 'react-device-detect';
 import styled from 'styled-components';
 import React from 'react';
 import { IconButton } from '../common/Button';
-import useENSAvatar from '../../../hooks/Guilds/ether-swr/ens/useENSAvatar';
+import useENSAvatar from '../../../hooks/Guilds/ens/useENSAvatar';
```
Comment (MiltonTulli) [inline src/old-components/Guilds/AddressButton/index.tsx:5]:
```suggestion
import useENSAvatar from 'hooks/Guilds/ens/useENSAvatar';
```
Let's try using absolute path import

File: src/old-components/Guilds/Avatar/ENSAvatar.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import styled from 'styled-components';
 import React from 'react';
-import useENSAvatar from '../../../hooks/Guilds/ether-swr/ens/useENSAvatar';
+import useENSAvatar from '../../../hooks/Guilds/ens/useENSAvatar';
```
Comment (MiltonTulli) [inline src/old-components/Guilds/Avatar/ENSAvatar.tsx:3]:
```suggestion
import useENSAvatar from 'hooks/Guilds/ens/useENSAvatar';

```

File: src/hooks/Guilds/ens/useENS.test.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      isError: false,
+    }));
+    const { address, name } = useENS(mockEnsName);
+    expect(address).toMatchInlineSnapshot(`""`);
+    expect(name).toMatchInlineSnapshot(`"wagmi.eth"`);
+  });
+
+  it('should return a valid address', () => {
+    const mockAddress: any = '0x0000000000000000000000000000000000000000';
+    const mockEnsName: any = 'wagmi.eth';
```
[the remark this answers is not in the document]
Comment (Kenny-Gin1) [reply src/hooks/Guilds/ens/useENS.test.ts:36]: Didn't work. Something weird with `wagmi` use of Typescript. It's not the best

File: src/hooks/Guilds/ens/useENS.test.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    isError: false,
+  }),
+}));
+
+describe('useENS', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+  it('should return the resolved ens name', () => {
+    const mockEnsName: any = 'wagmi.eth';
```
[the remark this answers is not in the document]
Comment (Kenny-Gin1) [reply src/hooks/Guilds/ens/useENS.test.ts:22]: Didn't work. There's something wrong with `wagmi's` use of TypeScript.

File: src/hooks/Guilds/ens/useAddressFromENSName.test.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+describe('useAddressFromENSName', () => {
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+  it('should return the address from the ens name', () => {
+    const { ensAddress, isError, isLoading } =
+      useAddressFromENSName('wagmi.eth');
+    expect(ensAddress).toBe('0x0000000000000000000000000000000000000000');
+    expect(isError).toBe(false);
+    expect(isLoading).toBe(false);
```
[the remark this answers is not in the document]
Comment (Kenny-Gin1) [reply src/hooks/Guilds/ens/useAddressFromENSName.test.ts:20]: Yeah, the tests are only there for functionality or to explain to the user how the hooks work.

**Comment by dcrescimbeni** [comment]
Everything looks good :star2: 

Waiting for QA results to approve.

**Review by Filipv95 — APPROVED** [review]

**Review by rossneilson — DISMISSED** [review]

**Review by rossneilson — APPROVED** [review]

**Review by dcrescimbeni — APPROVED** [review]

**Review by mprasanjith — APPROVED** [review]

**Pull request merged**